### PR TITLE
Pychromecast7

### PR DESCRIPTION
**Pull request by theychx**
Ongoing......... :tv:

### CODE REVIEW

**Comment by skorokithakis** [comment]
Thank you for this. While you're developing, can you add function types where practical? Currently we don't indicate what we accept or return, and it's making development hard. I tried to add some but discovery isn't working for me and couldn't figure out what functions were supposed to return.

File: catt/controllers.py [hunk 1]
```diff
@@ -41,23 +43,43 @@ def __init__(self, app_name, app_id, supported_device_types):
...
+        devices, zconf = get_chromecasts()
+    return [get_ccinfo(d, zconf) for d in devices]
+
+
+def get_ccinfo(device, zconf=None):
+    if zconf is None:
+        ip, port = device.host, device.port
+    else:
+        service_name = (
+            "-".join(device.model_name.split()) + "-" + str(device.uuid).replace("-", "") + "." + GOOGLE_ZCONF
```
Comment (theychx) [inline catt/controllers.py:62]: As per [this](https://github.com/home-assistant-libs/pychromecast/issues/381#issuecomment-651945670) comment, the `service_name` should be available from the `Chromecast` object, but I'm struggling to find just that.

**Comment by skorokithakis** [comment]
To clarify, I'm not saying add types to everything, just whatever function you happen to touch and is easy.

File: catt/controllers.py (hunk 1 above)
Comment (skorokithakis) [reply catt/controllers.py:62]: I'll give it a shot, I'll reconfigure my network so I can do discovery and look for it.

**Comment by theychx** [comment]
Gotcha. Basic casting should be working. I just need to get the auxilliary stuff up to date.

**Comment by skorokithakis** [comment]
I resolved some conflicts and will look for the `service_name` now.

**Comment by theychx** [comment]
There's a new `pychromecast` PR: https://github.com/home-assistant-libs/pychromecast/pull/385
I suggest we wait and see how that pans out, before putting more effort into this.

**Comment by skorokithakis** [comment]
Ah, agreed. I will do some work adding types to master in the mean time. What do you think regarding using PyChromecast's socket to talk to the Chromecast? They seemed to indicate that the socket is better for when groups change their leader, etc.

**Comment by theychx** [comment]
> What do you think regarding using PyChromecast's socket to talk to the Chromecast

That shouldn't be neccesary. Our current crude approach in `get_cast` already handles this (i.e. connection to the old leader fails, we do discovery and find the new leader). If anything, we should use the new `get_listed_chromecasts` to do "re-discovery" in these cases.

**Comment by skorokithakis** [comment]
@theychx the PR was merged, which is certainly interesting. Let's hope they keep that method of discovery and don't break it later...

**Pull request closed without merging**